server/service/appconfig.go

```diff
@@ -201,6 +202,10 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte) (*fleet.AppCo
 	}
 
 	oldSmtpSettings := appConfig.SMTPSettings
+	oldAgentOptions := ""
+	if appConfig.AgentOptions != nil {
+		oldAgentOptions = string(*appConfig.AgentOptions)
+	}
```

Comment:
I have been poking around this same area of the codebase, I like this approach of using variables instead of `appConfig` directly as it gets a bit confusing below once we unmarshal the new values into this struct.